infrastructure/environments/cloudformation/full/common/scaling.yaml

```diff
@@ -0,0 +1,116 @@
+AWSTemplateFormatVersion: '2010-09-09'
+Description: ECS Service with Auto-Scaling
+
+
+Parameters:
+  DaemonServiceName:
+    Type: String
+    Description: Name for the Dagster Daemon Service
+  DaemonServiceArn:
+    Type: String
+    Description: ARN for the Dagster Daemon Service
+  DagitServiceName:
+    Type: String
+    Description: Name for the Dagit Service
+  DagitServiceARN:
+    Type: String
+    Description: ARN for the Dagit Service
+  CodeServerServiceName:
+    Type: String
+    Description: Name for the Dagit Service
+  CodeServerServiceARN:
+    Type: String
+    Description: ARN for the Dagit Service
+  ECSClusterName:
+    Type: String
+    Description: Name for the ECS Dagster Service
+
+Resources:
+  FonsDagsterScalingLambdaRole:
+    Type: AWS::IAM::Role
+    Properties:
+      AssumeRolePolicyDocument:
+        Version: '2012-10-17'
+        Statement:
+          - Effect: Allow
+            Principal:
+              Service: lambda.amazonaws.com
+            Action: 'sts:AssumeRole'
+      Policies:
+        - PolicyName: ECSScalingPolicy
+          PolicyDocument:
+            Version: '2012-10-17'
+            Statement:
+              - Effect: Allow
+                Action:
+                  - ecs:UpdateService
+                Resource:
+                  - !Ref DaemonServiceARN
+                  - !Ref DagitServiceARN
+                  - !Ref CodeServerServiceARN
+
+  FonsScalingUpLambda:
+    Type: AWS::Lambda::Function
+    Properties:
+      Code:
+        S3Bucket: !Ref LambdaCodeBucket
+        S3Key: scaling-up-lambda.zip
+      Role: !GetAtt MyScalingLambdaRole.Arn
+      Handler: index.lambda_handler
+      Runtime: python3.11
+      Timeout: 300
+      Environment:
+        Variables:
+          ECS_CLUSTER_NAME: !Ref ECSClusterName
+          ECS_DAGIT_SERVICE_NAMES: !Ref DagitServiceName
+          ECS_DAEMON_SERVICE_NAMES: !Ref DaemonServiceName
+          ECS_CODE_SERVER_SERVICE_NAMES: !Ref CodeServerServiceName
+
+  FonsScalingDownLambda:
+    Type: AWS::Lambda::Function
+    Properties:
+      Code:
+        S3Bucket: !Ref LambdaCodeBucket
+        S3Key: scaling-down-lambda.zip
+      Role: !GetAtt MyScalingLambdaRole.Arn
+      Handler: index.lambda_handler
+      Runtime: python3.11
+      Timeout: 300
+      Environment:
+        Variables:
+          ECS_CLUSTER_NAME: !Ref ECSClusterName
+          ECS_DAGIT_SERVICE_NAMES: !Ref DagitServiceName
+          ECS_DAEMON_SERVICE_NAMES: !Ref DaemonServiceName
+          ECS_CODE_SERVER_SERVICE_NAMES: !Ref CodeServerServiceName
+
```

Comment:
I'm new to CloudFormation so this may be off base. Is there an opportunity here to DRY the lambdas, passing the desired count as a parameter? Or does that feel like over-engineering?

Reply:
Great point! I did think that actually and originally designed it to be just that, but then we'd need two schedules that would surface different values to a single lambda.  For instance, at 11pm it would set the count as 1, and at 5am it would set the count as 0. The problem comes...where are those values stored and sent to the lambda? In other words, what sends the parameter to this function?

There's a lot of ways to do this (API Call, step functions, etc), but all ways I've found involve adding more services to the tech stack which I'm reluctant to do for cost control reasons. These may longer-term involve those, but I opted for the simplest solution first.  Happy to rewrite if you think there's a simple/cheap way to unify these.

Reply:
Ah gotcha, it doesn't feel _ideal_ but could they be passed as environment variables here? It would mean the lambda is defined twice in the infrastructure, but the python behind it is not. Also could theoretically be useful if we ever want the desired count to be more than 1 (although perhaps that is not a likely scenario).

Or, I've never used this before, but looks like there is an [InputTransformer](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-events-rule-inputtransformer.html) that allows the passing of data to the lambda via `event`. ChatGPT seems to think that's a reasonable way forward, but whether that falls into it's 60% hit rate remains to be seen 😅

Reply:
An example according to chatgpt: 
![image](https://github.com/user-attachments/assets/c7b9b4bc-8192-41c9-a69c-711dd4c7e376)
![image](https://github.com/user-attachments/assets/9e834682-68d5-415e-bb1d-3be49a85ac47)